codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocket08FrameDecoder.java

```diff
@@ -409,17 +418,26 @@ private void unmask(ByteBuf frame) {
     }
 
     private void protocolViolation(ChannelHandlerContext ctx, String reason) {
-        protocolViolation(ctx, new CorruptedFrameException(reason));
+        protocolViolation(ctx, WebSocketCloseStatus.PROTOCOL_ERROR, reason);
+    }
+
+    private void protocolViolation(ChannelHandlerContext ctx, WebSocketCloseStatus status, String reason) {
+        protocolViolation(ctx, new CorruptedWebSocketFrameException(status, reason));
     }
 
-    private void protocolViolation(ChannelHandlerContext ctx, CorruptedFrameException ex) {
+    private void protocolViolation(ChannelHandlerContext ctx, CorruptedWebSocketFrameException ex) {
         state = State.CORRUPT;
-        if (ctx.channel().isActive()) {
+        if (ctx.channel().isActive() && config.closeOnProtocolViolation()) {
             Object closeMessage;
             if (receivedClosingHandshake) {
                 closeMessage = Unpooled.EMPTY_BUFFER;
             } else {
-                closeMessage = new CloseWebSocketFrame(1002, null);
+                WebSocketCloseStatus closeStatus = ex.getCloseStatus();
+                String reasonText = ex.getMessage();
+                if (reasonText == null || reasonText.isEmpty()) {
+                    reasonText = closeStatus.reasonText();
```

Comment:
I think we should only "override" if null

Reply:
fixed